src/protocols/MeterS0.cpp

```diff
@@ -105,34 +109,50 @@ int MeterS0::close() {
 
 ssize_t MeterS0::read(std::vector<Reading> &rds, size_t n) {
 
-	struct timeval time1;
-	struct timeval time2;
+	struct timeval time_now;
 	char buf[8];
 
 	/* clear input buffer */
 	tcflush(_fd, TCIOFLUSH);
 
-	/* blocking until one character/pulse is read */
-	if (::read(_fd, buf, 8) < 1) return 0;
-	gettimeofday(&time1, NULL);
+	// wait for very first impulse
+	if (!_impulseReceived) {
+		// blocking until one character/pulse is read
+		if (::read(_fd, buf, 8) < 1) return 0;
+		gettimeofday(&_time_last, NULL);
+
+		_impulseReceived = true;
+
+		// store timestamp
+		rds[0].identifier(new StringIdentifier("Impulse"));
+		rds[0].time(_time_last);
+		rds[0].value(1);
+
+		return 1;
+	}
+
+	// blocking until one character/pulse is read
 	if (::read(_fd, buf, 8) < 1) return 0;
-	gettimeofday(&time2, NULL);
+	gettimeofday(&time_now, NULL);
+
+	// _time_last is initialized at this point
+	double t1 = _time_last.tv_sec + _time_last.tv_usec / 1e6;
+	double t2 = time_now.tv_sec + time_now.tv_usec / 1e6;
+	double value = 3600000 / ((t2-t1) * _resolution);
 
-	double t1 = time1.tv_sec + time1.tv_usec / 1e6;
-	double t2 = time2.tv_sec + time2.tv_usec / 1e6;
-	double value = ( 3600000 ) / ( (t2-t1) * _resolution ) ;
+	memcopy(&_time_last, &time_now, sizeof(timeval));
```

Comment:
see travis-ci. memcopy is a typo -> memcpy (and shouldn't it be sizeof( struct timeval)?)